source/incqueryserver-jupyter/iqs_jupyter/authentication.py

```diff
@@ -64,26 +81,42 @@ def __init__(
             initial_password=defaults.default_IQS_password,
             use_oidc=defaults.default_use_oidc,
             token=defaults.default_IQS_token,
+            initial_auth_header_name=defaults.default_IQS_auth_header_name,
+            initial_auth_header_value=defaults.default_IQS_auth_header_value,
+            use_auth_header=defaults.default_use_auth_header,
             address_label='Address:',
+            use_user_pw_label='Login with Username & Password',
             user_label='User:',
-            oicd_checkbox_label='Use OpenID Connect',
             password_label='Password:',
+            oidc_checkbox_label='Use OpenID Connect',
+            token_label='OIDC Token:',
             label_text="IQS API Access Point",
+            auth_header_checkbox_label='Use Authentication Header',
+            auth_header_name_label='Header Name:',
+            auth_header_value_label='Header Value:',
             login_button=True,
             auto_display=defaults.default_auto_display
     ):
-        self.ask_for_user_pw = ask_for_user_pw
-        self.token = token
 
+        self.ask_for_user_pw_checkbox = widgets.Checkbox(value=ask_for_user_pw, description=use_user_pw_label)
         self.address_field = widgets.Text(value=initial_address, description=address_label)
         self.user_field = widgets.Text(value=initial_user, description=user_label)
         self.password_field = widgets.Password(value=initial_password, description=password_label)
-        self.oicd_checkbox = widgets.Checkbox(value=use_oidc, description=oicd_checkbox_label)
+
+        self.oidc_checkbox = widgets.Checkbox(value=use_oidc, description=oidc_checkbox_label)
+        self.token_field = widgets.Text(value=token, description=token_label)
+
+        self.auth_header_checkbox = widgets.Checkbox(value=use_auth_header, description=auth_header_checkbox_label)
+        self.auth_header_name_field = widgets.Text(value=initial_auth_header_name, description=auth_header_name_label)
+        self.auth_header_value_field = widgets.Text(value=initial_auth_header_value, description=auth_header_value_label)
 
         self.iqs_client = None
 
         if ask_for_user_pw:
```

Comment:
??? if there is a `ask_for_user_pw_checkbox` why do you use its (possibly obsolete) initial value, as opposed to its actual value? will the set of displayed widgets change if I check/uncheck the checkbox? 

Same with `auth_header_checkbox`